File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/KillChildAgents.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

/**
 * KillChildAgents - A new agent has connected to the simulator . . . make sure that any old child cameras are blitzed
 *
 * <p>Template: {@code KillChildAgents Low 242 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
public class KillChildAgents extends SLMessage {
    public IDBlock IDBlock_Field;

    /** Block IDBlock, Single. */
    public static class IDBlock {
        public UUID AgentID; // LLUUID
    }

    public KillChildAgents() {
        this.zeroCoded = false;
        this.IDBlock_Field = new IDBlock();
    }

    @Override
    public int CalcPayloadSize() {
        return 20;
    }

    @Override
    public void Handle(SLMessageHandler messageHandler) {
        messageHandler.HandleKillChildAgents(this);
    }

    @Override
    public void PackPayload(ByteBuffer byteBuffer) {
        // Message number: Low 242 (KillChildAgents).
        byteBuffer.putShort((short) 0xFFFF);
        byteBuffer.put((byte) 0x00);
        byteBuffer.put((byte) 0xF2);
        packUUID(byteBuffer, this.IDBlock_Field.AgentID);
    }

    @Override
    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.IDBlock_Field.AgentID = unpackUUID(byteBuffer);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/KillChildAgents.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer
import java.util.UUID

/**
 * KillChildAgents - A new agent has connected to the simulator . . . make sure that any old child cameras are blitzed
 *
 * <p>Template: {@code KillChildAgents Low 242 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
open class KillChildAgents : SLMessage() {
    @JvmField var IDBlock_Field: IDBlock = IDBlock()

    /** Block IDBlock, Single. */
    open class IDBlock {
        @JvmField var AgentID: UUID? = null
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 20
    }

    override fun Handle(messageHandler: SLMessageHandler) {
        messageHandler.HandleKillChildAgents(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        // Message number: Low 242 (KillChildAgents).
        byteBuffer.putShort(0xFFFF.toShort())
        byteBuffer.put((0x00).toByte())
        byteBuffer.put((0xF2).toByte())
        packUUID(byteBuffer, IDBlock_Field.AgentID)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        IDBlock_Field.AgentID = unpackUUID(byteBuffer)
    }
}
